finance/contracts/src/main/kotlin/net/corda/finance/contracts/asset/Obligation.kt

```diff
@@ -461,18 +489,42 @@ class Obligation<P : Any> : Contract {
                 "an acceptable contract is attached" using acceptableContract
             }
 
-            // That would pass this check. Ensuring they do not is best addressed in the transaction generation stage.
-            val assetStates = tx.outputsOfType<FungibleAsset<*>>()
-            // do not need to check acceptable asset as sometimes, the obligation with same template can be netted to 0
-//            val acceptableAssetStates = assetStates.filter {
-//                // Restrict the states to those of the correct issuance definition (this normally
-//                // covers issued product and obligor, but is opaque to us)
-//                it.amount.token in template.acceptableIssuedProducts
-//            }
-            // Catch that there's nothing useful here, so we can dump out a useful error
-            requireThat {
-                "there are fungible asset state outputs" using (assetStates.isNotEmpty())
-//                "there are defined acceptable fungible asset states" using (acceptableAssetStates.isNotEmpty())
+            // only inspect product if asset input match obligation term's product
+            val product = template.acceptableIssuedProducts.first().product
```

Comment:
we can not assume there will be only one acceptableIssueProduct.

Reply:
Done